// angular-twitter-timeline/esm2020/lib/angular-twitter-timeline.component.mjs
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./angular-twitter-timeline.service";
export class AngularTwitterTimelineComponent {
    constructor(element, twitterTimelineService) {
        this.element = element;
        this.twitterTimelineService = twitterTimelineService;
        this.defaultOpts = {
            tweetLimit: 5
        };
        this.defaultData = {
            sourceType: 'url',
            url: 'https://twitter.com/mustafaer_dev',
            screenName: 'Mustafa ER'
        };
    }
    ngOnChanges() {
        if (this.data && this.data.sourceType) {
            switch (this.data.sourceType) {
                case 'url':
                    delete this.defaultData.screenName;
                    break;
                case 'profile':
                    delete this.defaultData.url;
                    break;
                default:
                    break;
            }
            this.loadTwitterWidget();
        }
    }
    loadTwitterWidget() {
        this.twitterTimelineService
            .loadScript()
            .subscribe({
            next: () => {
                let nativeElement = this.element.nativeElement;
                nativeElement.innerHTML = "";
                window['twttr']
                    .widgets
                    .createTimeline({ ...this.defaultData, ...this.data }, nativeElement, { ...this.defaultOpts, ...this.opts })
                    .then(() => {
                })
                    .catch((error) => console.error(error));
            },
            error: (error) => console.error(error),
        });
    }
}
AngularTwitterTimelineComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: AngularTwitterTimelineComponent, deps: [{ token: i0.ElementRef }, { token: i1.AngularTwitterTimelineService }], target: i0.ɵɵFactoryTarget.Component });
AngularTwitterTimelineComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.0.4", type: AngularTwitterTimelineComponent, selector: "angular-twitter-timeline", inputs: { data: "data", opts: "opts" }, usesOnChanges: true, ngImport: i0, template: ``, isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: AngularTwitterTimelineComponent, decorators: [{
            type: Component,
            args: [{ selector: 'angular-twitter-timeline', template: `` }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i1.AngularTwitterTimelineService }]; }, propDecorators: { data: [{
                type: Input
            }], opts: [{
                type: Input
            }] } });
