// angular-twitter-timeline/esm2020/lib/angular-twitter-timeline.service.mjs
import { Injectable } from '@angular/core';
import { Observable } from "rxjs";
import * as i0 from "@angular/core";
export class AngularTwitterTimelineService {
    constructor() {
        this.TWITTER_SCRIPT_ID = 'twitter-wjs';
        this.TWITTER_WIDGET_URL = 'https://platform.twitter.com/widgets.js';
    }
    loadScript() {
        return new Observable((observer) => {
            this.startScriptLoad();
            window['twttr'].ready((twttr) => {
                observer.next(twttr);
                observer.complete();
            });
        });
    }
    startScriptLoad() {
        window['twttr'] = (function (d, s, id, url) {
            let script, firstScriptEl = d.getElementsByTagName(s)[0], twitterScript = window['twttr'] || {};
            if (d.getElementById(id)) {
                return twitterScript;
            }
            script = d.createElement(s);
            script.id = id;
            script.src = url;
            firstScriptEl.parentNode.insertBefore(script, firstScriptEl);
            twitterScript._e = [];
            twitterScript.ready = function (f) {
                twitterScript._e.push(f);
            };
            return twitterScript;
        }(document, 'script', this.TWITTER_SCRIPT_ID, this.TWITTER_WIDGET_URL));
    }
}
AngularTwitterTimelineService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: AngularTwitterTimelineService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
AngularTwitterTimelineService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: AngularTwitterTimelineService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: AngularTwitterTimelineService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
